### DISCUSSION TITLE
Accessing enfugue with the browser

### DISCUSSION THREAD

**Discussion by lennartbrandin** [painebenjamin/app.enfugue.ai#24] [General]
When accessing enfugue you will notice two different domains:
- `https://my.enfugue.ai` -> 52.219.92.88
- `https://app.enfugue.ai` -> 127.0.0.1

In the end `https://my.enfugue.ai` is a redirect to `https://app.enfugue.ai:45554`.

Im wondering why this public redirect exists, for me it seems to only obfuscate the documentation on how to access the app.
I would prefer if the documentation just referred to `http://127.0.0.1:45554`.

**Comment by painebenjamin** [comment]
Hello! Thank you for the comment. Many people are seeing the redirects as superfluous complication - in a way it is, but it is also not. Let me explain:

1. Some advanced browser features are locked behind SSL. This is a purposeful security measure done by browser developers to try and keep unsecured connections from accessing API's that allow interacting with your computer outside of the boundaries of the web browser, such as the Clipboard API for writing things directly to your operating system clipboard.
2. That means that in order to use these and _not_ require users to go mucking about in their `chrome://flags` page, the server must be serving over HTTPS and use a valid SSL certificate - and one cannot secure by IP address, only by domain.

With those reasons together, I knew Enfugue needed to be able to server over SSL, so I architected this circuitous method. The `my.enfugue.ai` part is purely convenience for people who aren't familiar with typing ports, but it seems it may be more confusing than it is helpful, so I may just remove that entirely. `my.enfugue.ai` isn't actually a website, it's just a proxy S3 bucket I registered with AWS.

With all that being said, though, there was another power user here who communicated their frustrations with this, and I explained to them how it can be configured to turn on/off SSL, change domain, ports, etc. The details are in [this issue](https://github.com/painebenjamin/app.enfugue.ai/issues/18). I've added some command-line utilities to make configuring easy in the next iteration.
